SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp

```diff
@@ -46,9 +51,32 @@ void USpatialGDKEditorSettings::PostInitProperties()
 	PlayInSettings->PostEditChange();
 	PlayInSettings->SaveConfig();
 
+	SetRuntimeWorkerTypes();
 	SafetyCheckSpatialOSDirectoryPaths();
 }
 
+void USpatialGDKEditorSettings::SetRuntimeWorkerTypes()
+{
+	TSet<FName> WorkerTypes;
+
+	for (const FWorkerTypeLaunchSection WorkerLaunch : LaunchConfigDesc.Workers)
+	{
+		if (WorkerLaunch.WorkerTypeName != NAME_None)
+		{
+			WorkerTypes.Add(WorkerLaunch.WorkerTypeName);
+		}
+	}
+
+	USpatialGDKSettings* RuntimeSettings = GetMutableDefault<USpatialGDKSettings>();
+	if (RuntimeSettings != nullptr)
+	{
+		RuntimeSettings->WorkerTypes.Empty();
+		RuntimeSettings->WorkerTypes.Append(WorkerTypes);
```

Comment:
What's the reason for emptying then appending? Will this deallocate the array then reallocate when appending? Can we just copy in that case or call `RuntimeSettings->WorkerTypes.Empty(WorkerTypes.Num());`?

Reply:
True. Not super critical code since it's an editor bit, but I guess we might as well.